mava/components/tf/architectures/networked.py

```diff
@@ -38,20 +38,21 @@ def __init__(
         network_spec: Dict[str, List[str]],
         observation_networks: Dict[str, snt.Module],
         policy_networks: Dict[str, snt.Module],
-        shared_weights: bool = False,
+        agent_net_keys: Dict[str, str],
     ):
         super().__init__(
             environment_spec=environment_spec,
             observation_networks=observation_networks,
             policy_networks=policy_networks,
-            shared_weights=shared_weights,
+            agent_net_keys=agent_net_keys,
         )
 
         self._network_spec = network_spec
 
-        if self._shared_weights:
+        if len(set(self._agent_net_keys.values())) < len(self._agent_net_keys.values()):
```

Comment:
@DriesSmit Can you give me an example of when this would be true?

Reply:
> Thanks @DriesSmit , this looks really good! Much nicer way of handling agent networks  
> 
> I know you made this backward compatible, but just checking that you checked this doesn't cause issues with the examples? Also, could you please also update `make_custom_network` and this line `network_factory = lp_utils.partial_kwargs(maddpg.make_default_networks,shared_weights=False)` in `quickstart.ipynb`.

I checked a few different examples throughout Mava and everything seems to still be training :+1:  I updated the notebook.

Reply:
> @DriesSmit Can you give me an example of when this would be true?

When agents share networks. Then the set of possible networks are less than the number of agents.